File: agent/models/realtime_data.py
from pydantic import BaseModel, field_validator
from datetime import datetime

class realtime_data(BaseModel):
    heart_rate: int
    spo2: int
    stress_level: int
    steps: int
    calories_burned: int
    timestamp: datetime

    @field_validator("timestamp", mode="before")
    def parse_ts(cls, v):
        return datetime.fromtimestamp(v / 1000)

File: backend/src/models/RealtimeData.js


import mongoose from "mongoose";

const realtimeDataSchema = new mongoose.Schema({
    heart_rate: { type: Number, required: true },
    spo2: { type: Number },
    stress_level: { type: Number },
    steps: { type: Number },
    calories_burned: { type: Number },
    timestamp: { type: Date, required: true },
}, {
    timestamps: true,
    collection: "realtime_data"   // 🔥 important
});

// Default export
const RealtimeData = mongoose.model("RealtimeData", realtimeDataSchema);

export default RealtimeData;
